### DISCUSSION TITLE
Clean up attribute inheritance for theming

### DISCUSSION THREAD

**Discussion by jkrumbiegel** [JuliaPlots/Makie.jl#4196] [Ideas]
With the new figure and the larger role of Axis types our old model fails in unexpected ways, so we should think about how to clean it up.

The premise should be that (almost) everything is themeable, every plot type, every Layoutable like Axis, Colorbar, etc.

Some places which involve attributes that we can manipulate are:
- the `const` default minimal theme
- the current_theme via `set_theme!`
- the Figure has an as of now completely unused `attributes` field
- the Layoutables directly at creation
- Plot types directly at creation
- Scenes underlying Figure and Axis

Because layoutables used to be put directly into a scene, they want to inherit possible attributes from that scene. This is still the case even with Figure. Actually, the keywords passed at creation of the figure are currently just forwarded to `fig.scene` and the layoutables grab them from there.

Layoutables check for an entry with their own type name in the theme like `Axis = (xticks = ...` or `Colorbar...`.

Plots had that behavior at some point but it doesn't seem to work anymore to theme `scatter` with a keyword in the theme. Something's broken there.

Basically, we need to come up with some graph that describes how attributes flow from the top to the bottom. Any ideas?

**Comment by ffreyer** [comment]
Can this be closed?
